tectonic/core-bridge.c

```diff
@@ -137,6 +137,26 @@ ttstub_issue_error(const_string format, ...)
     TGB->issue_error(TGB->context, error_buf);
 }
 
+PRINTF_FUNC(2,3) int
+ttstub_fprintf(rust_output_handle_t handle, const_string format, ...)
+{
+    static char fprintf_buf[BUF_SIZE] = "";
+    va_list ap;
+
+    va_start(ap, format);
+    int len = vsnprintf(fprintf_buf, BUF_SIZE, format, ap);
+    va_end(ap);
+
+    if (len >= BUF_SIZE) {
+        _tt_abort("tt_fprintf: buffer overflow");
```

Comment:
This shouldn't abort on overflow. I think the best behavior is to just truncate the buffer and proceed, as in [dpx-error.c](https://github.com/tectonic-typesetting/tectonic/blob/master/tectonic/dpx-error.c#L71-L74) (the other functions in this file should also be more careful about ensuring that the string is definitely terminated, but that's something for another PR).